CONTRIBUTING.md

```diff
@@ -1,24 +1,181 @@
+# Wrong
+old_translate_fn = QtWidgets.QApplication.translate
+
+def translate(context, key, disambiguation=None, encoding=None, n=0):
+    return old_translate_fn(context, key, disambiguation, n)
+
+# Overwrite original with an incompatible version
+QtWidgets.QApplication.translate = staticmethod(translate)
+```
+
+```python
+# Right
+...
+QtWidgets.QApplication.translate_ = staticmethod(translate)
+```
 
-Development for this project takes place in individual forks. The parent project ever only contains a single branch, a branch containing the latest working version of the project.
+<br>
+
+##### Don't get smart
+
+At the end of the day, Qt.py is a middle-man. It delegates request you make to the appropriate reciever, such as PySide2. There are many ways in which this delegation can go, and Python - being a capable dynamic programming language - can easily make something like this simple high-level goal into a tangled mess of automatic and optimised nest of bugs.
+
+That is why delegation is a direct mapping between source -> target binding, with little or no code in between.
+
+<br>
+
+##### No bugs
+
+This may seem like an impossible requirement, but hear me out. Bugs stem from implementations. Ergo, if there are no implementations, there can be no bugs.
+
+Qt.py merely maps one binding to look like another. Implementations are left to the source developers.
+
+```python
+# Wrong
+def QWidget(source_binding, *args, **kwargs):
+    # Potential bug 1
+    if kwargs["__special_option"] == 0x1336:
+        kwargs["__magic"] = 0x1337
+
+    # Potential bug 2
+    return getattr(source_binding, "QWidget")(*args, *kwargs)
+
+# Potential bug 3
+QtWidgets.QWidget = lambda *args, **kwargs: QWidget(PySide, *args, **kwargs)
+```
+
+```python
+# Right
+QtWidgets.QWidget = QtGui.QWidget  # No bugs
+```
+
+<br>
+
+## How can I contribute?
+
+Contribution comes in many flavours, some of which is simply notifying us of problems or successes, so we know what to change and not to change.
+
+### Reporting bugs
 
 Bugreports must include:
 
 1. Description
 2. Expected results
 3. Short reproducible
- 
+
+### Suggesting enhancements
+
 Feature requests must include:
 
 1. Goal (what the feature aims to solve)
 2. Motivation (why *you* think this is necessary)
 3. Suggested implementation (psuedocode)
 
 Questions may also be submitted as issues.
+
+### Pull requests
+
+Code contributions are made by (1) forking this project and (2) making a modification to it. Your code will be reviewed and merged once it:
+
+1. Does something useful
+1. Is up to par with surrounding code
+
+The parent project ever only contains a single branch, a branch containing the latest working version of the project.
+
+We understand and recognise that "forking" and "pull-requests" can be a daunting aspect for a beginner, so don't hesitate to ask. A pull-request should normally follow an issue where you elaborate on your desires; this is also a good place to ask about these things.
+
```

Comment:
To pull requests, I would add a modest note on that it may take some discussing back and forth before we can make a decision on whether to accept the PR or if it should be modified... add a test etc. And that this process can take a while. Despite all of this, we encourage PRs and are super happy to have people interested in contributing with their time and efforts.